src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4468,34 +4468,46 @@ FindReferencedTableColumn(Expr *columnExpression, List *parentQueryList, Query *
 		return;
 	}
 
-	/*
-	 * We currently don't support finding partition keys in the subqueries
-	 * that reference outer subqueries. For example, in correlated
-	 * subqueries in WHERE clause, we don't support use of partition keys
-	 * in the subquery that is referred from the outer query.
-	 */
-	if (candidateColumn->varlevelsup > 0 && !skipOuterVars)
+
+	if (candidateColumn->varlevelsup > 0)
 	{
-		int parentQueryIndex = list_length(parentQueryList) -
-							   candidateColumn->varlevelsup;
-		if (!(IsIndexInRange(parentQueryList, parentQueryIndex)))
+		if (!skipOuterVars)
```

Comment:
Minor: I find single level indentation simpler to follow, what do you think?


```C
if (candidateColumn->varlevelsup > 0 && skipOuterVars)
{
/* caller doesn't want to process*/
   return;
}
else if (candidateColumn->varlevelsup > 0 && !skipOuterVars)
{
... do thnigs .. 
}
```

Reply:
umm, I didn't want to write `candidateColumn->varlevelsup > 0` twice.

What about this so that there is single level indendation for the main logic?:

```c
if (candidateColumn->varlevelsup > 0){
   if (skipOuterVars) {
      /* caller doesn't want to process*/
   return;
   }
   ... do things ....
}
```

Reply:
Yes, this looks nicer, thanks